## Some data analysis before the annotation process:

In [1]:
import os
import pandas as pd
import numpy as np

In [2]:
# Checking if we have same category names:
path = '../txtLAB/Doc Class 5.0/ECCO_TrainingData_Trial4/'

print(len(os.listdir(path)[2:]))

categories = ['Agriculture.csv', 'Botany.csv', 'Church.csv', 'Commerce.csv', 
              'Drama.csv', 'Fiction.csv', 'History.csv', 'History_Natural.csv', 'Law.csv', 'Mathematics.csv', 
              'Medicine.csv', 'Physics.csv', 'Poetry.csv', 'Politics.csv', 'Rhetoric.csv', 'Sermons.csv', 
              'Travels.csv']
print(set(categories) == set(os.listdir(path)[2:]))

17
True


#### Creating a table for all 17 classes:

In [3]:
df_s = []
for cat in os.listdir(path)[2:]:
    df = pd.read_csv(path+cat)
    df['Class Label'] = cat[:-4]
    print(cat)
    print(df.shape)
    df_s.append(df)

Agriculture.csv
(1403, 10)
Botany.csv
(494, 10)
Church.csv
(5077, 10)
Commerce.csv
(2530, 10)
Drama.csv
(4708, 10)
Fiction.csv
(4557, 10)
History.csv
(5619, 10)
History_Natural.csv
(501, 10)
Law.csv
(4402, 10)
Mathematics.csv
(1282, 10)
Medicine.csv
(4043, 10)
Physics.csv
(852, 10)
Poetry.csv
(6159, 10)
Politics.csv
(4182, 10)
Rhetoric.csv
(1334, 10)
Sermons.csv
(7410, 10)
Travels.csv
(2683, 10)


In [4]:
table1_all_17 = pd.concat(df_s, ignore_index=True)
print(table1_all_17.nunique())
table1_all_17.head()

Filename          57236
DocumentID        57236
ESTC_ID           49018
Date                124
Title             46190
Vol_Number          124
Author            13332
Imprint           41750
Field_Headings    24966
Class Label          17
dtype: int64


Filename  DocumentID  ESTC_ID  Date  \
0  0604100200.xml   604100200  T122642  1771   
1  0623400500.xml   623400500  T118425  1771   
2  0651800800.xml   651800800  T040653  1794   
3  0680600100.xml   680600100  T040647  1799   
4  0810200300.xml   810200300  T138847  1767   

                                               Title Vol_Number  \
0  A six weeks tour, through the southern countie...          0   
1  An abridgment of the six weeks, and six months...          0   
2  General view of the agriculture of the county ...          0   
3  General view of the agriculture of the county ...          0   
4  Certain ancient tracts concerning the manageme...          0   

                       Author  \
0               Young, Arthur   
1               Young, Arthur   
2  Foot, Peter, land-surveyor   
3               Young, Arthur   
4                        Anon   

                                             Imprint  \
0            Dublin : printed for J. Milliken, 1771.   
1               Dublin : printed by S. Powell, 1771.   
2            London : printed by John Nichols, 1794.   
3  London : printed by W. Bulmer and Co. for G. N...   
4  London : printed for C. Bathurst, at the Cross...   

                                      Field_Headings  Class Label  
0           Agriculture, Wales, Agriculture, England  Agriculture  
1                               Agriculture, England  Agriculture  
2  Agriculture, England, London Region, London (E...  Agriculture  
3  Agriculture, England, Lincolnshire, Lincolnshi...  Agriculture  
4  Farm management, Surveying, Early works to 180...  Agriculture

In [5]:
table1_all_17['Class Label'].value_counts()

Sermons            7410
Poetry             6159
History            5619
Church             5077
Drama              4708
Fiction            4557
Law                4402
Politics           4182
Medicine           4043
Travels            2683
Commerce           2530
Agriculture        1403
Rhetoric           1334
Mathematics        1282
Physics             852
History_Natural     501
Botany              494
Name: Class Label, dtype: int64

In [6]:
table1_all_17.to_csv('/Users/dh_lab_05/Desktop/FINAL_Labelling/training_data.csv', index=None)

In [7]:
check = pd.read_csv('./training_data.csv')
check.head()

Filename  DocumentID  ESTC_ID  Date  \
0  0604100200.xml   604100200  T122642  1771   
1  0623400500.xml   623400500  T118425  1771   
2  0651800800.xml   651800800  T040653  1794   
3  0680600100.xml   680600100  T040647  1799   
4  0810200300.xml   810200300  T138847  1767   

                                               Title Vol_Number  \
0  A six weeks tour, through the southern countie...          0   
1  An abridgment of the six weeks, and six months...          0   
2  General view of the agriculture of the county ...          0   
3  General view of the agriculture of the county ...          0   
4  Certain ancient tracts concerning the manageme...          0   

                       Author  \
0               Young, Arthur   
1               Young, Arthur   
2  Foot, Peter, land-surveyor   
3               Young, Arthur   
4                        Anon   

                                             Imprint  \
0            Dublin : printed for J. Milliken, 1771.   
1               Dublin : printed by S. Powell, 1771.   
2            London : printed by John Nichols, 1794.   
3  London : printed by W. Bulmer and Co. for G. N...   
4  London : printed for C. Bathurst, at the Cross...   

                                      Field_Headings  Class Label  
0           Agriculture, Wales, Agriculture, England  Agriculture  
1                               Agriculture, England  Agriculture  
2  Agriculture, England, London Region, London (E...  Agriculture  
3  Agriculture, England, Lincolnshire, Lincolnshi...  Agriculture  
4  Farm management, Surveying, Early works to 180...  Agriculture

In [8]:
check.nunique()

Filename          57236
DocumentID        57236
ESTC_ID           49018
Date                124
Title             46190
Vol_Number          124
Author            13332
Imprint           41750
Field_Headings    24966
Class Label          17
dtype: int64

#### Creating a table for ALL the data we have:

In [9]:
path = '../txtLAB/Koustuv_ECCO_Metadata/'

df_s2 = []
for cat in os.listdir(path):
    df = pd.read_csv(path+cat, delimiter='\t')
    df['TableName'] = cat[:-4]
    print(cat)
    print(df.shape)
    df_s2.append(df)

Manifest_GenRef.tsv
(4182, 10)
Manifest_HistAndGeo.tsv
(14570, 10)
Manifest_Law.tsv
(9966, 10)
Manifest_LitAndLang1.tsv
(31980, 10)
Manifest_LitAndLang2.tsv
(11249, 10)
Manifest_MedSciTech.tsv
(12448, 10)
Manifest_RelandPhil.tsv
(39079, 10)
Manifest_SSAndFineArt.tsv
(31450, 10)


In [10]:
table2_all_metadata = pd.concat(df_s2, ignore_index=True)
print(table2_all_metadata.nunique())
table2_all_metadata.head()

Filename          154924
DocumentID        154924
ESTC_ID           136164
Date                 142
Title             126134
Vol_Number           186
Author             25459
Imprint           105324
Field_Headings     60827
TableName              8
dtype: int64


Filename  DocumentID  ESTC_ID  Date  \
0  0031300100.xml    31300100  T021625  1736   
1  0031300200.xml    31300200  T013049  1787   
2  0031300300.xml    31300300  T057382  1800   
3  0031300400.xml    31300400  T012488  1787   
4  0031300500.xml    31300500  W029739  1773   

                                               Title Vol_Number  \
0  Bibliotheca topographica Anglicana: or, a new ...          0   
1  A catalogue of books printed for, and sold by ...          0   
2  Rules of a reading-society, established April ...          0   
3  Rules for regulating the subscription library ...          0   
4  A catalogue of books, imported and to be sold ...          0   

           Author                                            Imprint  \
0   Worrall, John  London : printed for J. Worrall at the Dove in...   
1  Dilly, Charles                               [London], s.n, 1787.   
2            Anon  London : printed by H.D. Steel, No. 51, Lothbu...   
3            Anon    Stamford : printed by Newcomb and Peat, [1787].   
4     Knox, Henry               [Boston : Sold by Henry Knox, 1773].   

                                      Field_Headings        TableName  
0  Books, Prices, Catalogs, Booksellers', Great B...  Manifest_GenRef  
1        Catalogs, Booksellers', Early works to 1800  Manifest_GenRef  
2  [London Reading Society], Rules and practice, ...  Manifest_GenRef  
3  [Stamford Subscription Library], Rules and pra...  Manifest_GenRef  
4  Booksellers and bookselling, Massachusetts, Bo...  Manifest_GenRef

#### This means, I actually have to label 97688 (154924 - 57236) files.
#### Creating a CSV that has only the metadata I need to actually label:

In [11]:
filenames_already_labelled = table1_all_17['Filename'].tolist()
print(len(filenames_already_labelled))

57236


In [12]:
to_label = table2_all_metadata.loc[~table2_all_metadata['Filename'].isin(filenames_already_labelled)]

In [13]:
to_label.nunique()

Filename          97688
DocumentID        97688
ESTC_ID           87270
Date                131
Title             81957
Vol_Number          146
Author            18892
Imprint           69271
Field_Headings    41218
TableName             8
dtype: int64

In [14]:
to_label.to_csv('./fnames_to_label.csv', index=None)

In [15]:
temp = pd.read_csv('./fnames_to_label.csv')
temp.head()

Filename  DocumentID  ESTC_ID  Date  \
0  0031300100.xml    31300100  T021625  1736   
1  0031300200.xml    31300200  T013049  1787   
2  0031300300.xml    31300300  T057382  1800   
3  0031300400.xml    31300400  T012488  1787   
4  0031300500.xml    31300500  W029739  1773   

                                               Title Vol_Number  \
0  Bibliotheca topographica Anglicana: or, a new ...          0   
1  A catalogue of books printed for, and sold by ...          0   
2  Rules of a reading-society, established April ...          0   
3  Rules for regulating the subscription library ...          0   
4  A catalogue of books, imported and to be sold ...          0   

           Author                                            Imprint  \
0   Worrall, John  London : printed for J. Worrall at the Dove in...   
1  Dilly, Charles                               [London], s.n, 1787.   
2            Anon  London : printed by H.D. Steel, No. 51, Lothbu...   
3            Anon    Stamford : printed by Newcomb and Peat, [1787].   
4     Knox, Henry               [Boston : Sold by Henry Knox, 1773].   

                                      Field_Headings        TableName  
0  Books, Prices, Catalogs, Booksellers', Great B...  Manifest_GenRef  
1        Catalogs, Booksellers', Early works to 1800  Manifest_GenRef  
2  [London Reading Society], Rules and practice, ...  Manifest_GenRef  
3  [Stamford Subscription Library], Rules and pra...  Manifest_GenRef  
4  Booksellers and bookselling, Massachusetts, Bo...  Manifest_GenRef

In [16]:
temp.nunique()

Filename          97688
DocumentID        97688
ESTC_ID           87270
Date                131
Title             81957
Vol_Number          146
Author            18892
Imprint           69271
Field_Headings    41218
TableName             8
dtype: int64

In [17]:
# Sanity check: Checking if metadata had those 57k files used for training SVM:
print(len(filenames_already_labelled))
blah = table2_all_metadata.loc[table2_all_metadata['Filename'].isin(filenames_already_labelled)]
print(blah.nunique())
blah.head()

57236
Filename          57236
DocumentID        57236
ESTC_ID           49018
Date                124
Title             46189
Vol_Number          123
Author            13333
Imprint           41748
Field_Headings    24971
TableName             7
dtype: int64


Filename  DocumentID  ESTC_ID  Date  \
4182  0002200101.xml     2200101  T093173  1781   
4183  0002200102.xml     2200102  T093173  1781   
4185  0002200300.xml     2200300  T141125  1702   
4186  0002200400.xml     2200400  T097665  1791   
4187  0002200501.xml     2200501  T042143  1747   

                                                  Title Vol_Number  \
4182  The field of Mars: being an alphabetical diges...   Volume 1   
4183  The field of Mars: being an alphabetical diges...   Volume 2   
4185  An introduction to the history of the principa...          0   
4186  The spirit of general history, in a series of ...          0   
4187  An historical review of the transactions of Eu...   Volume 1   

                               Author  \
4182                             Anon   
4183                             Anon   
4185  Pufendorf, Samuel, Freiherr von   
4186        Thomson, George, Reverend   
4187                    Boyse, Samuel   

                                                Imprint  \
4182  London : printed for J. Macgowan, No. 27, Pate...   
4183  London : printed for J. Macgowan, No. 27, Pate...   
4185  London : printed for Tho. Newborough at the Go...   
4186  Carlisle : printed by F. Jollie, and sold by B...   
4187  Reading : printed by and for D. Henry in Friar...   

                                  Field_Headings            TableName  
4182  Military history, Military art and science  Manifest_HistAndGeo  
4183  Military history, Military art and science  Manifest_HistAndGeo  
4185                             Europe, History  Manifest_HistAndGeo  
4186        History, Modern, History, Philosophy  Manifest_HistAndGeo  
4187               Europe, History, 18th century  Manifest_HistAndGeo